test/unit/controllers/consolidations/AssociateDucrControllerSpec.scala

```diff
@@ -57,76 +57,64 @@ class AssociateDucrControllerSpec extends ControllerSpec {
     super.afterEach()
   }
 
-  "Associate DUCR GET" should {
+  "Associate Ducr controller" should {
 
-    "throw incomplete application when cache empty" in {
+    "return OK (200)" when {
 
-      withCaching(MucrOptions.formId, None)
+      "display page method is invoked and Mucr Options page has data" in {
 
-      assertThrows[IncompleteApplication] {
-        await(controller.displayPage()(getRequest()))
-      }
-    }
-
-    "return Ok for GET request" in {
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
 
-      withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        val result = controller.displayPage()(getRequest())
 
-      val result = controller.displayPage()(getRequest())
-
-      status(result) must be(OK)
+        status(result) must be(OK)
+      }
     }
-  }
-
-  "Associate DUCR POST" should {
 
-    "throw incomplete application when cache empty" in {
+    "throw an IncompleteApplication exception" when {
 
-      withCaching(MucrOptions.formId, None)
+      "display page method is invoked and Mucr Options page is not filled" in {
+        withCaching(MucrOptions.formId, None)
 
-      assertThrows[IncompleteApplication] {
-        await(controller.submit()(postRequest(Json.obj())))
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
       }
-    }
-
-    "display an error for a missing DUCR" in {
 
-      withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+      "Mucr Options page is not filled during saving with incorrect form" in {
```

Comment:
In both test cases the word "filled" is misleading, as it suggests that Mucr Options should be entered on this page. Maybe you could refactor the name to explicitly say that Mucr Options should/should not be in cache?

Reply:
Fixed